core/src/main/java/org/axonframework/commandhandling/disruptor/FirstLevelCache.java

```diff
@@ -0,0 +1,95 @@
+package org.axonframework.commandhandling.disruptor;
+
+import org.axonframework.eventsourcing.EventSourcedAggregate;
+
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Works as an in-memory cache to keep a weak reference to the cached object,
+ * which then allows the garbage collector to remove an object from memory once it isn't needed
+ * anymore.
+ * <p>
+ * A {@link HashMap} doesn't help here since it will keep hard delegate for key and
+ * value objects. A {@link WeakHashMap} doesn't either, because it keeps weak delegate to the
+ * key objects, but we want to track the value objects.
+ * <p>
+ * This implementation which delegates to a {@link Map} uses a {@link WeakReference} for the values. Once the
+ * garbage collector decides it wants to finalize a value, it will be removed from the
+ * map automatically.
+ * <p>
+ * This implementation is heavily inspired by http://www.java2s.com/Code/Java/Collections-Data-Structure/WeakValueHashMap.htm
+ *
+ * @param <T> the type of the aggregate root
```

Comment:
Due to the added comment, I assume this will be part of either 3.4 or 3.3.1. Might be nice to add the `@since` tag in that case as well.